scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py

```diff
@@ -0,0 +1,218 @@
+
+    def config_lvm(self):
+        node_name = self.conf_store.get(self.index, f'cluster>server[{self.server_id}]>hostname', default_val=None)
+        node_name = node_name.split('.')[0]
+        metadata_device = self.conf_store.get(self.index,
+        f'cluster>server[{self.server_id}]>storage>metadata_devices', default_val=None)
+        is_physical = True if self.conf_store.get(self.index, f'cluster>server[{self.server_id}]>node_type', default_val=None) == "HW" else False
+
+        self.create_lvm(node_name, metadata_device[0], is_physical)
+
+    def config_motr(self):
+        cmd = f"/opt/seagate/cortx/motr/libexec/motr_cfg.sh"
+        self.execute_command(cmd, 180)
+
+    def test_motr(self):
+        cmd = f"uname -r"
+        op = self.execute_command(cmd, 180)
+        kernel_ver = op.replace('\n', '')
+
+        self.validate_file(f"/lib/modules/{kernel_ver}/kernel/fs/motr/m0tr.ko")
+
+        self.validate_file("/etc/sysconfig/motr")
+
+    def validate_file(self, file):
+        if not os.path.exists(file):
+            print(f"[ERR] {file} does not exist.")
+            sys.exit(1)
+        else:
+            print(f"[MSG] {file} exists.")
+
+    def test_lnet(self):
+        missing_pkgs = []
+        LNET_CONF_FILE="/etc/modprobe.d/lnet.conf"
+        search_lnet_pkgs = ["kmod-lustre-client", "lustre-client"]
+
+        # Check missing luster packages
+        cmd = "rpm -qa | grep lustre"
+        temp =  self.execute_command(cmd, 180)
+        lustre_pkgs = list(filter(None, temp.split("\n")))
+        for pkg in search_lnet_pkgs:
+            found = False
+            for lustre_pkg in lustre_pkgs:
+                if pkg in lustre_pkg:
+                    found = True
+                    break;
+            if found == False:
+                missing_pkgs.append(pkg)
+        if missing_pkgs:
+            print("Missing pkgs ={}".format(missing_pkgs))
+
+        # Check for lnet config file
+        if os.path.exists(LNET_CONF_FILE):
+            with open(LNET_CONF_FILE) as fp:
+                line = fp.readline()
+                while line:
+                    tokens = line.split(' ')
+                    # Get lnet iface
+                    cmd = 'echo \'{}\' | cut -d "(" -f2 | cut -d ")" -f1'.format(tokens[2])
+                    device = self.execute_command(cmd, 180)
+                    device = device.strip('\n')
+                    print("iface:{}".format(device))
+
+                    # Get ip of iface
+                    cmd = "ifconfig {} | awk \'/inet /\'".format(device)
+                    ipconfig_op = self.execute_command(cmd, 180)
+                    ip = list(ipconfig_op.split())[1]
+                    print("ip = {}".format(ip))
+
+                    # Ping ip
+                    cmd = "ping -c 3 {}".format(ip)
+                    op = self.execute_command(cmd, 180)
+                    print(op)
+                    line = fp.readline()
+            fp.close()
```

Comment:
fp.close() is not requried when using with

Reply:
Done 1a235dd